### ISSUE TITLE
Find All References is really slow for this project

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#5485]
See the repro details at https://github.com/microsoft/vscode-cpptools/issues/5376 . That issue got fixed, but the performance is unusually slow. I wasn't able to get Find All References to work in VS for comparison. There may be some performance bug. There might be a recursive includes and tag parser cause for the slowness in addition to IntelliSense parsing. The unification work may or may not improve this, but it would be a good test case to see if performance got better or worse after the unification work changes.

**Comment by axiqia** [comment]
VScode is a very useful tool for me. It plays an important role in my work and research, I use it to write code and papers. I have been developing Linux kernel with VScode for a long time. 

As we all know, Linux kernel is a quite large project. The `compile_commands.json` generated by this [repo](https://github.com/amezin/vscode-linux-kernel) has 8317 entries. When I jump to the definition of a function (`funcA`) for the first time, it really takes **a long time**. VScode logs "[Warning] Some references may be missing, because workspace parsing was incomplete when Find All References was started." many times. Thankfully, the following same jumps to `funcA` are very quick.

However, if I jump to `funcA` after another function, it will take a long time again. May we could cache the previous parsed results to figure this out?

**Comment by sean-mcmanus** [comment]
Are you trying to do a Find All References or Go to Definition? The warning message means the workspace hasn't finished parsing the files.

**Comment by axiqia** [comment]
I'm trying to **Go to Definition**. I expect that we could go to definition as soon as possible, it's a frequent action when we are reading the source code of a project. Why does the workspace need to parse files again and again instead of only once?

**Comment by sean-mcmanus** [comment]
@axiqia The original issue is on Find All References...you may wan to file a new issue. It sounds like the Go to Definition is falling back to a Find All References for some reason. You can try changing the editor.gotoLocation.alternativeDefinition setting to change that behavior. Also, the workspace should only need to re-parse if there was a potential database corruption previously, such as what may happen if you close VS Code while a database write is occurring. Setting C_Cpp.loggingLevel to "Debug" will log a message when that happens.
